crates/nu-cli/src/repl.rs

```diff
@@ -422,141 +421,185 @@ fn loop_iteration(
         file!(),
         line!(),
         column!(),
-        use_color,
+        ctx.use_color,
     );
 
-    *entry_num += 1;
+    *ctx.entry_num += 1;
 
     start_time = std::time::Instant::now();
     line_editor = line_editor.with_transient_prompt(transient_prompt);
-    let input = line_editor.read_line(nu_prompt);
+    let input = line_editor.read_line(ctx.nu_prompt);
     let shell_integration = config.shell_integration;
 
     match input {
         Ok(Signal::Success(s)) => {
             let hostname = System::host_name();
             let history_supports_meta = matches!(
-                engine_state.history_config().map(|h| h.file_format),
+                ctx.engine_state.history_config().map(|h| h.file_format),
                 Some(HistoryFileFormat::Sqlite)
             );
 
             if history_supports_meta {
-                prepare_history_metadata(&s, &hostname, engine_state, &mut line_editor)?;
+                if let Err(e) =
+                    prepare_history_metadata(&s, &hostname, ctx.engine_state, &mut line_editor)
+                {
+                    warn!("Could not prepare history metadata: {e}");
+                }
             }
 
             // Right before we start running the code the user gave us, fire the `pre_execution`
             // hook
             if let Some(hook) = config.hooks.pre_execution.clone() {
                 // Set the REPL buffer to the current command for the "pre_execution" hook
-                let mut repl = engine_state.repl_state.lock().expect("repl state mutex");
+                let mut repl = ctx
+                    .engine_state
+                    .repl_state
+                    .lock()
+                    .expect("repl state mutex");
                 repl.buffer = s.to_string();
                 drop(repl);
 
-                if let Err(err) =
-                    eval_hook(engine_state, stack, None, vec![], &hook, "pre_execution")
-                {
-                    report_error_new(engine_state, &err);
+                if let Err(err) = eval_hook(
+                    ctx.engine_state,
+                    ctx.stack,
+                    None,
+                    vec![],
+                    &hook,
+                    "pre_execution",
+                ) {
+                    report_error_new(ctx.engine_state, &err);
                 }
             }
 
-            let mut repl = engine_state.repl_state.lock().expect("repl state mutex");
+            let mut repl = ctx
+                .engine_state
+                .repl_state
+                .lock()
+                .expect("repl state mutex");
             repl.cursor_pos = line_editor.current_insertion_point();
             repl.buffer = line_editor.current_buffer_contents().to_string();
             drop(repl);
 
             if shell_integration {
-                run_ansi_sequence(PRE_EXECUTE_MARKER)?;
+                if let Err(e) = run_ansi_sequence(PRE_EXECUTE_MARKER) {
```

Comment:
I think it would be fine to change `run_ansi_sequence` to not return an `Result` and instead do the `warn!` there. This should eliminate a bunch of the repeat error handling. With this, I think you might be able to change the signatures of some of the other functions to not return `Result`s as well. E.g., `do_shell_integration_finalize_command` only has one other inner source of error (`coerce_into_string`), so the error handling could be moved inside there too.

Reply:
Thanks! I think `do_run_cmd` can also be made to return `Reedline` instead of `Result<Reedline>`. This should unlock a few other simplifications.

Reply:
done!